File: backend/ModuMessenger/src/main/java/com/example/modumessenger/member/entity/Role.java
package com.example.modumessenger.member.entity;

import lombok.Getter;

import java.util.Arrays;
import java.util.NoSuchElementException;

@Getter
public enum Role {
    ROLE_ADMIN("ROLE_ADMIN"),
    ROLE_MEMBER("ROLE_USER");

    final private String roleName;

    Role(String roleName) {
        this.roleName = roleName;
    }

    public boolean isCorrectName(String name) {
        return name.equalsIgnoreCase(this.roleName);
    }

    public  static Role getRoleByName(String roleName) {
        return Arrays.stream(Role.values())
                .filter(r -> r.isCorrectName(roleName))
                .findFirst()
                .orElseThrow(() -> new NoSuchElementException("존재하지 않는 권한 종류 입니다."));
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/member/entity/Role.kt
package com.example.memberservice.member.entity

enum class Role(val roleName: String) {
    ROLE_ADMIN("ROLE_ADMIN"),
    ROLE_MEMBER("ROLE_USER"),
    ;

    fun isCorrectName(name: String): Boolean = name.equals(roleName, ignoreCase = true)

    companion object {
        @JvmStatic
        fun getRoleByName(roleName: String): Role =
            entries.firstOrNull { it.isCorrectName(roleName) }
                ?: throw NoSuchElementException("존재하지 않는 권한 종류 입니다.")
    }
}
